api/src/utils/media/csv/validation/csv-row-validator.ts

```diff
@@ -98,18 +97,18 @@ export const getCodeValueFieldsValidator = (config?: ColumnCodeValidatorConfig):
       // compare allowed code values as lowercase strings
       const allowedCodeValuesLowerCase: (string | number)[] = [];
       const allowedCodeValues = config.column_code_validator.allowed_code_values.map((allowedCode) => {
-        allowedCodeValuesLowerCase.push(safeToLowerCase(allowedCode.name));
+        allowedCodeValuesLowerCase.push(String(safeToLowerCase(allowedCode.name)));
```

Comment:
If `allowedCode.name` isn't a String, the `safeToLowerCase` will not convert it to lowercase. So the end result of this operation is always a String, but it is not guarenteed to be lowercase. Since we're always attempting to cast `allowedCode.name` to String, should we swap the order of operations to assure that it always gets converted to lowercase?

Reply:
good catch. Updated both instances of this casting to cast to string first, then try to lowercase it